File: generated-code/example/swift/Sources/TransGenTest/Enumeration.swift
/// Example enumeration
public enum Enumeration: Int32 {
    /// First option
    case valueOne = 0

    /// Second option
    case valueTwo = 1

    /// Read Enumeration from input stream
    static func readFrom<S: InputStream>(_ stream: S) -> Enumeration {
        return Enumeration(rawValue: stream.readInt32())!
    }

    /// Write Enumeration to output stream
    func writeTo<S: OutputStream>(_ stream: S) {
        stream.writeInt32(rawValue)
    }
}

File: generated-code/example/haskell/FileReadWrite/src/Model/Enumeration.hs
module Model.Enumeration where

import Prelude hiding (id)
import qualified Trans
import Trans (Trans)
import Data.Int

-- | Example enumeration
data Enumeration
    -- | First option
    = ValueOne
    -- | Second option
    | ValueTwo
    deriving (Eq, Ord, Show)

instance Trans Enumeration where
    read = do
        tag :: Int32 <- Trans.read
        return $ case tag of
            0 -> ValueOne
            1 -> ValueTwo

    write ValueOne =
        Trans.write (0 :: Int32)
    write ValueTwo =
        Trans.write (1 :: Int32)